Moving to Cli 2.0.2 broke something

Tried running `build.cmd run` on latest head, and failed.
I could see it was doing some dotnet --version magic, and downloading the `2.0.2` version. But when trying to build I get the following error:

```
C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
ncyResolution.targets(323,5): error MSB4018: The "ReportAssetsLogMessages" task failed unexpectedly. [C
:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj] [C:\Projects\SAFE-BookStore\test\ServerT
ests\ServerTests.fsproj]
C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
ncyResolution.targets(323,5): error MSB4018: System.IO.FileLoadException: Could not load file or assemb
ly 'NuGet.ProjectModel, Version=4.4.0.3, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. Could not f
ind or load a specific file. (Exception from HRESULT: 0x80131621) [C:\Projects\SAFE-BookStore\test\Serv
erTests\ServerTests.fsproj] [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj]
C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
ncyResolution.targets(323,5): error MSB4018: File name: 'NuGet.ProjectModel, Version=4.4.0.3, Culture=n
eutral, PublicKeyToken=31bf3856ad364e35' [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fspro
j] [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj]
C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
ncyResolution.targets(323,5): error MSB4018: at Microsoft.NET.Build.Tasks.ReportAssetsLogMessages.Execu
teCore() [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj] [C:\Projects\SAFE-BookStore\t
est\ServerTests\ServerTests.fsproj]
C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
ncyResolution.targets(323,5): error MSB4018: at Microsoft.NET.Build.Tasks.TaskBase.Execute() [C:\Projec
ts\SAFE-BookStore\test\ServerTests\ServerTests.fsproj] [C:\Projects\SAFE-BookStore\test\ServerTests\Ser
verTests.fsproj]
C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
ncyResolution.targets(323,5): error MSB4018: at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Bui
ld.BackEnd.ITaskExecutionHost.Execute() [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj
] [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj]
C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
ncyResolution.targets(323,5): error MSB4018: at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiate
dTask>d__25.MoveNext() [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj] [C:\Projects\SA
FE-BookStore\test\ServerTests\ServerTests.fsproj]
C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
ncyResolution.targets(323,5): error MSB4018: [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.f
sproj] [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj]
C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
ncyResolution.targets(323,5): error MSB4018: [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.f
sproj] [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj]
C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
ncyResolution.targets(165,5): error MSB4018: The "CheckForTargetInAssetsFile" task failed unexpectedly.
 [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj] [C:\Projects\SAFE-BookStore\test\Serv
erTests\ServerTests.fsproj]
C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
ncyResolution.targets(165,5): error MSB4018: System.IO.FileLoadException: Could not load file or assemb
ly 'NuGet.Frameworks, Version=4.4.0.3, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. Could not fin
d or load a specific file. (Exception from HRESULT: 0x80131621) [C:\Projects\SAFE-BookStore\test\Server
Tests\ServerTests.fsproj] [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj]
C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
ncyResolution.targets(165,5): error MSB4018: File name: 'NuGet.Frameworks, Version=4.4.0.3, Culture=neu
tral, PublicKeyToken=31bf3856ad364e35' [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj]
 [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj]
C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
ncyResolution.targets(165,5): error MSB4018: at Microsoft.NET.Build.Tasks.CheckForTargetInAssetsFile.Ex
ecuteCore() [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj] [C:\Projects\SAFE-BookStor
e\test\ServerTests\ServerTests.fsproj]
C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
ncyResolution.targets(165,5): error MSB4018: at Microsoft.NET.Build.Tasks.TaskBase.Execute() [C:\Projec
ts\SAFE-BookStore\test\ServerTests\ServerTests.fsproj] [C:\Projects\SAFE-BookStore\test\ServerTests\Ser
verTests.fsproj]
C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
ncyResolution.targets(165,5): error MSB4018: at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Bui
ld.BackEnd.ITaskExecutionHost.Execute() [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj
] [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj]
C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
ncyResolution.targets(165,5): error MSB4018: at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiate
dTask>d__25.MoveNext() [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj] [C:\Projects\SA
FE-BookStore\test\ServerTests\ServerTests.fsproj]
C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
ncyResolution.targets(165,5): error MSB4018: [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.f
sproj] [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj]
C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
ncyResolution.targets(165,5): error MSB4018: [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.f
sproj] [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj]
```

Looks like some dependency resolution is not doing what it is supposed to.

Tested commits backwards, and the last one working is https://github.com/SAFE-Stack/SAFE-BookStore/commit/9e01a314cee7237421850d8a5db3c94e5250f381 (last one before moving to 2.0.2

Running windows 10 falls creators. Doing dotnet --version I have 2.0.0 as the system level sdk version.

I have never seen this. @cartermp?

Am 27.10.2017 12:33 schrieb "Vidar Løvbrekke Sømme" <
notifications@github.com>:

> Tried running build.cmd run on latest head, and failed.
> I could see it was doing some dotnet --version magic, and downloading the
> 2.0.2 version. But when trying to build I get the following error:
>
> C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
> ncyResolution.targets(323,5): error MSB4018: The "ReportAssetsLogMessages" task failed unexpectedly. [C
> :\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj] [C:\Projects\SAFE-BookStore\test\ServerT
> ests\ServerTests.fsproj]
> C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
> ncyResolution.targets(323,5): error MSB4018: System.IO.FileLoadException: Could not load file or assemb
> ly 'NuGet.ProjectModel, Version=4.4.0.3, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. Could not f
> ind or load a specific file. (Exception from HRESULT: 0x80131621) [C:\Projects\SAFE-BookStore\test\Serv
> erTests\ServerTests.fsproj] [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj]
> C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
> ncyResolution.targets(323,5): error MSB4018: File name: 'NuGet.ProjectModel, Version=4.4.0.3, Culture=n
> eutral, PublicKeyToken=31bf3856ad364e35' [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fspro
> j] [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj]
> C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
> ncyResolution.targets(323,5): error MSB4018: at Microsoft.NET.Build.Tasks.ReportAssetsLogMessages.Execu
> teCore() [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj] [C:\Projects\SAFE-BookStore\t
> est\ServerTests\ServerTests.fsproj]
> C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
> ncyResolution.targets(323,5): error MSB4018: at Microsoft.NET.Build.Tasks.TaskBase.Execute() [C:\Projec
> ts\SAFE-BookStore\test\ServerTests\ServerTests.fsproj] [C:\Projects\SAFE-BookStore\test\ServerTests\Ser
> verTests.fsproj]
> C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
> ncyResolution.targets(323,5): error MSB4018: at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Bui
> ld.BackEnd.ITaskExecutionHost.Execute() [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj
> ] [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj]
> C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
> ncyResolution.targets(323,5): error MSB4018: at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiate
> dTask>d__25.MoveNext() [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj] [C:\Projects\SA
> FE-BookStore\test\ServerTests\ServerTests.fsproj]
> C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
> ncyResolution.targets(323,5): error MSB4018: [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.f
> sproj] [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj]
> C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
> ncyResolution.targets(323,5): error MSB4018: [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.f
> sproj] [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj]
> C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
> ncyResolution.targets(165,5): error MSB4018: The "CheckForTargetInAssetsFile" task failed unexpectedly.
>  [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj] [C:\Projects\SAFE-BookStore\test\Serv
> erTests\ServerTests.fsproj]
> C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
> ncyResolution.targets(165,5): error MSB4018: System.IO.FileLoadException: Could not load file or assemb
> ly 'NuGet.Frameworks, Version=4.4.0.3, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. Could not fin
> d or load a specific file. (Exception from HRESULT: 0x80131621) [C:\Projects\SAFE-BookStore\test\Server
> Tests\ServerTests.fsproj] [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj]
> C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
> ncyResolution.targets(165,5): error MSB4018: File name: 'NuGet.Frameworks, Version=4.4.0.3, Culture=neu
> tral, PublicKeyToken=31bf3856ad364e35' [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj]
>  [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj]
> C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
> ncyResolution.targets(165,5): error MSB4018: at Microsoft.NET.Build.Tasks.CheckForTargetInAssetsFile.Ex
> ecuteCore() [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj] [C:\Projects\SAFE-BookStor
> e\test\ServerTests\ServerTests.fsproj]
> C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
> ncyResolution.targets(165,5): error MSB4018: at Microsoft.NET.Build.Tasks.TaskBase.Execute() [C:\Projec
> ts\SAFE-BookStore\test\ServerTests\ServerTests.fsproj] [C:\Projects\SAFE-BookStore\test\ServerTests\Ser
> verTests.fsproj]
> C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
> ncyResolution.targets(165,5): error MSB4018: at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Bui
> ld.BackEnd.ITaskExecutionHost.Execute() [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj
> ] [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj]
> C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
> ncyResolution.targets(165,5): error MSB4018: at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiate
> dTask>d__25.MoveNext() [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj] [C:\Projects\SA
> FE-BookStore\test\ServerTests\ServerTests.fsproj]
> C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
> ncyResolution.targets(165,5): error MSB4018: [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.f
> sproj] [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj]
> C:\Users\vidar\AppData\Local\dotnetcore\sdk\2.0.2\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDepende
> ncyResolution.targets(165,5): error MSB4018: [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.f
> sproj] [C:\Projects\SAFE-BookStore\test\ServerTests\ServerTests.fsproj]
>
> Looks like some dependency resolution is not doing what it is supposed to.
>
> Tested commits backwards, and the last one working is 9e01a31
> <https://github.com/SAFE-Stack/SAFE-BookStore/commit/9e01a314cee7237421850d8a5db3c94e5250f381>
> (last one before moving to 2.0.2
>
> Running windows 10 falls creators. Doing dotnet --version I have 2.0.0 as
> the system level sdk version.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/SAFE-Stack/SAFE-BookStore/issues/231>, or mute the
> thread
> <https://github.com/notifications/unsubscribe-auth/AADgNIaUt5it91Zp7HMgVXmxraZmevLOks5swbEggaJpZM4QI3Fm>
> .
>

--- a/build.fsx
+++ b/build.fsx
@@ -216,7 +216,7 @@ Target "Run" (fun _ ->
             ExecProcess (fun info ->
                 info.FileName <- dotnetExePath
                 info.WorkingDirectory <- serverTestsPath
-                info.Arguments <- "watch msbuild /t:TestAndRun") TimeSpan.MaxValue
+                info.Arguments <- sprintf "watch msbuild /t:TestAndRun /p:DotNetHost=%s" dotnetExePath) TimeSpan.MaxValue
 
         if result <> 0 then failwith "Website shut down." }
 

--- a/test/ServerTests/ServerTests.fsproj
+++ b/test/ServerTests/ServerTests.fsproj
@@ -8,9 +8,12 @@
     <Compile Include="Tests.fs" />
     <Compile Include="Program.fs" />
   </ItemGroup>
+  <PropertyGroup>
+    <DotNetHost Condition="'$(DotNetHost)' == ''">dotnet</DotNetHost>
+  </PropertyGroup>
   <Target Name="TestAndRun">
-    <Exec Command="dotnet run" WorkingDirectory=".\" />
-    <Exec Command="dotnet run" WorkingDirectory="..\..\src\Server\" />
+    <Exec Command="$(DotNetHost) run" WorkingDirectory=".\" />
+    <Exec Command="$(DotNetHost) run" WorkingDirectory="..\..\src\Server\" />
   </Target>
   <ItemGroup>
     <ProjectReference Include="..\..\src\Server\Server.fsproj" />

